fcos-config.yaml

```diff
@@ -104,3 +131,45 @@ storage:
               exit 0
           fi
           nmcli device modify "$1" ipv4.method auto
+    - path: /usr/local/bin/verify-bearer-token
+          #!/bin/sh
+          # This file is part of CRC
+          # Note: this is a POC and should not be used in production
+          #
+          # Copyright (C) 2021 Red Hat, Inc.
+          set -eux
+          HOST="$1"
+          echo "Bearer auth attempt from $HOST" >&2
+          # FIXME: should be random
+          COOKIE=c1
+
+          # FIXME: compute frame length (in particular for different cookie length)
+          printf '61\n\n{ "command": "authorize", "cookie": "'$COOKIE'", "challenge": "*" }'
+          read framelen
+          read channel
+          [ "$channel" = "" ]
+          read -n $((framelen - 1)) response
+          echo "got response '$response'" >&2
+
+          # looks like '{"command":"authorize","cookie":"c1","response":"Bearer 123foo"}'
+          if [ "$(echo "$response" | jq -r .cookie)" != "$COOKIE" ]; then
+              # FIXME: send proper "problem" JSON
+              echo "bad cookie" >&2
+              exit 1
+          fi
+
+          if [ "$(echo "$response" | jq -r .response)" != "Bearer Y29yZQ==" ]; then
```

Comment:
this is the value that needs to be synced between the image and https://github.com/gbraad/podman-desktop/blob/9d13968a58eab54985fbabe23f58f0f3d9937d9a/main.js#L15